### PR TITLE
Cuz Im Happy - Backend

### PR DESCRIPTION
**Pull request by karanmann**
Happy Thought API
This project is made by me on my 19th week of Technigo Boot Camp. It was my first experience with Full Stack Development. Connecting my own Backend development to an existing Frontend project 

Documentation
# Show all data - 
https://happy-thoughts-dummy-api.herokuapp.com/thoughts

# GET & POST on - '/thoughts'

# POST on - '/thoughts/:id/like'
The problem
Since this assignment was a first in a lot of ways it was just a challenge to recall the previous knowledge. But eventually I was able to get it up and running

View it live
Front End : https://cuz-i-am-happy.netlify.app/ Back End / Deployment : https://happy-thoughts-dummy-api.herokuapp.com/thoughts

Tech Used
Node.js Express.js MongoDB + Atlas + Compass Mongoose Heroku

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -7,11 +7,28 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: [true, 'Message Required'],
+    minLength: [5, 'Message too short, minimum 5 characters'],
+    maxLength: [140, 'Message too long, maximum 140 characters']
```
Comment (sandra-shumanteva-kaevska) [inline server.js:15]: This should be minlength and maxlength without big L. If it is with big L is not working.

File: server.js [hunk 2]
```diff
@@ -7,11 +7,28 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
...
+  },
+  createdAt: {
+    type:Date,
+    default: () => Date.now()
```
Comment (sandra-shumanteva-kaevska) [inline server.js:19]: There is no need of function you can just do like:
 createdAt: {
    type: Date,
    default: Date.now
  }

File: server.js [hunk 3]
```diff
@@ -20,7 +37,44 @@ app.use(bodyParser.json())
...
+      {new: true}
+    )
+    res.status(200).json(updatedThought)
+  } catch (err) {
+    res.status(404).json({ 
+      message: 'Bad Request - Could not post like', 
+      error: err.errors
+    })
+  }
 })
```
Comment (sandra-shumanteva-kaevska) [inline server.js:79]: Great job Karan with the harts 💯

**Review by sandra-shumanteva-kaevska — CHANGES_REQUESTED** [review]
Great job Karan. 
Your code needs some fix to work properly. If I put 5 characters should the send Happy Thoughts button be clickable and post the message. Now it is not working.

File: server.js (hunk 3 above)
Comment (karanmann) [reply server.js:79]: Hi Sandra, Thank you for the review. I have made the requested changes and the problem of 5 characters was in the front end where it was 6 :) Now that is fixed too.

Cheers /K

**Review by KarolCislowski — APPROVED** [review]
Nice work 👍🏻

**Pull request closed without merging**